docs/sdk/src/reference_docs/frame_benchmarking_weight.rs

```diff
@@ -1,23 +1,211 @@
+//! the 20ms. In a benchmarked environment, it can examine the transactions for their upper bound,
+//! and include the ones that are known to fit based on the worst case.
+//!
+//! The benchmarking code can be written as a part of FRAME pallet, using the macros provided in
+//! [`frame_benchmarking`]. See any of the existing pallets in `polkadot-sdk`, or the pallets in our
+//! [`crate::polkadot_sdk::templates`] for examples.
+//!
+//! ## Weight
+//!
+//! Finally, [`sp_weights::Weight`] is the output of the benchmarking process. It is a
+//! two-dimensional data structure that demonstrates the resources consumed by a given block of
+//! code (for example, a transaction). The two dimensions are:
+//!
+//! * reference time: The time consumed in pico-seconds, on a reference hardware.
+//! * proof size: The amount of storage proof necessary to re-execute the block of code. This is
+//!   mainly needed for parachain <> relay-chain verification.
+//!
+//! ## How To Write Benchmarks: Worst Case
+//!
+//! The most important detail about writing benchmarking code is that it must be written such that
+//! it captures the worst case execution of any block of code.
+//!
+//! Consider:
+#![doc = docify::embed!("./src/reference_docs/frame_benchmarking_weight.rs", simple_transfer)]
+//!
+//! If this block of code is to be benchmarked, then the benchmarking code must be written such that
+//! it captures the worst case.
+//!
+//! ## Gluing Pallet Benchmarking with Runtime
+//!
+//! FRAME pallets are mandated to provide their own benchmarking code. Runtimes contain the
+//! boilerplate needed to run these benchmarking (see [Running Benchmarks
+//! below](#running-benchmarks)). The outcome of running these benchmarks are meant to be fed back
+//! into the pallet via a conventional `trait WeightInfo` on `Config`:
+#![doc = docify::embed!("src/reference_docs/frame_benchmarking_weight.rs", WeightInfo)]
+//!
+//! Then, individual functions of this trait are the final values that we assigned to the
+//! [`frame::pallet_macros::weight`] attribute:
+#![doc = docify::embed!("./src/reference_docs/frame_benchmarking_weight.rs", simple_transfer_2)]
+//!
+//! ## Manual Refund
+//!
+//! Back to the assumption of writing benchmarks for worst case: Sometimes, the pre-dispatch weight
+//! significantly differ from the post-dispatch actual weight consumed. This can be expressed with
+//! the following FRAME syntax:
+#![doc = docify::embed!("./src/reference_docs/frame_benchmarking_weight.rs", simple_transfer_3)]
+//!
+//! ## Running Benchmarks
+//!
+//! Two ways exist to run the benchmarks of a runtime.
+//!
+//! 1. The old school way: Most Polkadot-Sdk based nodes (such as the ones integrated in
+//!    [`templates`]) have an a `benchmark` subcommand integrated into themselves.
+//! 2. The more [`crate::reference_docs::omni_node`] compatible way of running the benchmarks would
+//!    be using [`frame-omni-bencher`] CLI, which only relies on a runtime.
```

Comment:
This link does not work locally for me. Probably needs to be added to worskpace deps (and polkadot-sdk/docs deps).

Reply:
ah yeah, for binaries I will use the crates.io link. They cannot be added as a dependnecy.